exp/MOM_compile.csh

```diff
@@ -156,10 +156,14 @@ if ( $cosima_version ) then
     set cppDefs = "$cppDefs -DCOSIMA_VERSION"
 endif
 
-# Build FMS.
-source ./FMS_compile.csh
-
-set includes = "-I$code_dir/shared/include -I$executable:h:h/lib_FMS -I$executable:h:h/lib_ocean"
+if ( ! $?SPACK_GTRACERS_EXTERNAL ) then
+    echo "Building type=$type with internal FMS"
+    source ./FMS_compile.csh
+    set includes = "-I$code_dir/shared/include -I$executable:h:h/lib_FMS -I$executable:h:h/lib_ocean"
+else
+    echo "Building type=$type with external FMS"
+    set includes = "-I$code_dir/shared/include -I$executable:h:h/lib_ocean"
+endif
```

Comment:
```suggestion
if ( ! $?SPACK_GTRACERS_EXTERNAL ) then
    echo "Building type=$type with internal FMS"
    source ./FMS_compile.csh
    set includes = "-I$code_dir/shared/include -I$executable:h:h/lib_FMS -I$executable:h:h/lib_ocean"
else
    echo "Building type=$type with external FMS"
    set includes = "-I$executable:h:h/lib_ocean"
endif
```
All of the `$code_dir/shared/include` directory is part of FMS so we shouldn't need to depend on it for the external FMS (spack) build. If it fails we have an unresolved dependency of some kind.